### PR TITLE
Organize target platforms

### PR DESCRIPTION
**Pull request by swankjesse**
I'm about to introduce new platforms and don't want to paste
the platforms list in a bunch of new places.

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]

**Comment by swankjesse** [comment]
Ugh, I couldn’t get buildSrc to work and it’s unclear why. Just lots of things where the Kotlin plugin is crashing on inconsistent versions of things. I moved the new code into a `build-support` directory instead.

File: build-support/src/main/kotlin/BuildSupport.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class BuildSupport : Plugin<Project> {
```
Comment (swankjesse) [inline build-support/src/main/kotlin/BuildSupport.kt:4]: I couldn’t figure out how to use functions from this module in my `build.gradle.kts` files without also implementing the plugin interface.

Comment (JakeWharton) [reply build-support/src/main/kotlin/BuildSupport.kt:4]: Super easy in Groovy Gradle 😁

Comment (swankjesse) [reply build-support/src/main/kotlin/BuildSupport.kt:4]: TRUE

**Pull request merged**